src/osd/osd_types.h

```diff
@@ -890,6 +891,74 @@ inline ostream& operator<<(ostream& out, const pool_snap_info_t& si) {
 
 
 /*
+ * pool_opts_t
+ *
+ * pool options.
+ */
+
+class pool_opts_t {
+public:
+  enum key_t {
+    SCRUB_MIN_INTERVAL,
+    SCRUB_MAX_INTERVAL,
+    DEEP_SCRUB_INTERVAL,
+  };
+
+  enum type_t {
+    STR,
+    INT,
+    DOUBLE,
+  };
+
+  struct opt_desc_t {
+    key_t key;
+    type_t type;
+
+    opt_desc_t(key_t k, type_t t) : key(k), type(t) {}
+  };
+
+  typedef boost::variant<std::string,int,double> value_t;
```

Comment:
Where do you think it might not work, and why other types (std::string, double) don't worry you?